interactors/transaction.go

```diff
@@ -0,0 +1,164 @@
+}
+
+type transactionInteractor struct {
+	Proxy
+	TxSigner
+	mutTxAccumulator           sync.RWMutex
+	millisecondsBetweenBunches uint32
+	txAccumulator              []*data.Transaction
+}
+
+// NewTransactionInteractor will create an interactor that extends the proxy functionality with some transaction-oriented functionality
+func NewTransactionInteractor(proxy Proxy, txSigner TxSigner) (*transactionInteractor, error) {
+	if check.IfNil(proxy) {
+		return nil, ErrNilProxy
+	}
+	if check.IfNil(txSigner) {
+		return nil, ErrNilTxSigner
+	}
+
+	return &transactionInteractor{
+		Proxy:    proxy,
+		TxSigner: txSigner,
+	}, nil
+}
+
+func (ti *transactionInteractor) SetMillisecondsBetweenBunches(millisecondsBetweenBunches uint32) {
+	atomic.StoreUint32(&ti.millisecondsBetweenBunches, millisecondsBetweenBunches)
+}
+
+// AddTransaction will add the provided transaction in the transaction accumulator
+func (ti *transactionInteractor) AddTransaction(tx *data.Transaction) {
+	if tx == nil {
+		return
+	}
+
+	ti.mutTxAccumulator.Lock()
+	ti.txAccumulator = append(ti.txAccumulator, tx)
+	ti.mutTxAccumulator.Unlock()
+}
+
+// PopAccumulatedTransactions will return the whole accumulated contents emptying the accumulator
+func (ti *transactionInteractor) PopAccumulatedTransactions() []*data.Transaction {
+	ti.mutTxAccumulator.Lock()
+	result := make([]*data.Transaction, len(ti.txAccumulator))
+	copy(result, ti.txAccumulator)
+	ti.txAccumulator = make([]*data.Transaction, 0)
+	ti.mutTxAccumulator.Unlock()
+
+	return result
+}
+
+// CreateTransaction assembles a transaction from the provided arguments
+func (ti *transactionInteractor) CreateTransaction(arg ArgCreateTransaction) *data.Transaction {
+	return &data.Transaction{
+		Nonce:     arg.Nonce,
+		Value:     arg.Value,
+		RcvAddr:   arg.RcvAddr,
+		SndAddr:   arg.SndAddr,
+		GasPrice:  arg.GasPrice,
+		GasLimit:  arg.GasLimit,
+		Data:      arg.Data,
+		Signature: arg.Signature,
+		ChainID:   arg.ChainID,
+		Version:   arg.Version,
+		Options:   arg.Options,
+	}
+}
+
+// ApplySignatureAndSender will apply the corresponding sender and compute the signature field upon the provided argument
+func (ti *transactionInteractor) ApplySignatureAndSender(skBytes []byte, arg ArgCreateTransaction) (ArgCreateTransaction, error) {
+	pkBytes, err := ti.TxSigner.GeneratePkBytes(skBytes)
+	if err != nil {
+		return ArgCreateTransaction{}, err
+	}
+
+	copyArg := arg
+	copyArg.Signature = ""
+	copyArg.SndAddr = core.AddressPublicKeyConverter.Encode(pkBytes)
+
+	unsignedMessaged, err := ti.createUnsignedMessage(copyArg)
+	if err != nil {
+		return ArgCreateTransaction{}, err
+	}
+
+	signature, err := ti.TxSigner.SignMessage(unsignedMessaged, skBytes)
+	if err != nil {
+		return ArgCreateTransaction{}, err
+	}
+
+	copyArg.Signature = hex.EncodeToString(signature)
+
+	return copyArg, nil
+}
+
+func (ti *transactionInteractor) createUnsignedMessage(arg ArgCreateTransaction) ([]byte, error) {
+	tx := ti.CreateTransaction(arg)
```

Comment:
add a `arg.Signature = ""` before just to be sure

Reply:
done